### PR TITLE
Implements player silencing on Unity

### PR DESCRIPTION
**Pull request by marc-n-dream**
This depends on https://github.com/AirConsole/airconsole-api/pull/76 and the follow up merge the release/1.9.0 branch into its master (with the parallel 1.8.0 and 1.9.0-alpha releases)

it will be merged along the related branches on documentation, appengine and airconsole-api

### CODE REVIEW

File: Assets/WebGLTemplates/AirConsole/airconsole-unity-plugin.js [hunk 1]
```diff
@@ -325,6 +325,8 @@ App.prototype.processUnityData = function (data) {
         this.airconsole.setCustomDeviceState(data.data);
     } else if (data.action == "setCustomDeviceStateProperty") {
         this.airconsole.setCustomDeviceStateProperty(data.key, data.value);
+    } else if (data.action === "enablePlayerSilencing"){
```
Comment (daniel-n-dream) [inline Assets/WebGLTemplates/AirConsole/airconsole-unity-plugin.js:328]:
```suggestion
    } else if (data.action === "enablePlayerSilencing") {
```

File: Assets/WebGLTemplates/AirConsole-2020/airconsole-unity-plugin.js [hunk 2]
```diff
@@ -354,6 +354,8 @@ App.prototype.processUnityData = function (data) {
         this.airconsole.setCustomDeviceState(data.data);
     } else if (data.action == "setCustomDeviceStateProperty") {
         this.airconsole.setCustomDeviceStateProperty(data.key, data.value);
+    } else if (data.action === "enablePlayerSilencing"){
```
Comment (daniel-n-dream) [inline Assets/WebGLTemplates/AirConsole-2020/airconsole-unity-plugin.js:357]:
```suggestion
    } else if (data.action === "enablePlayerSilencing") {
```

File: Assets/AirConsole/scripts/AirConsole.cs [hunk 3]
```diff
@@ -1930,6 +1939,26 @@ public class NotReadyException: SystemException {
...
 
 #endregion
+		
+#if UNITY_ANDROID || UNITY_EDITOR
+		/// <summary>
+		/// Enables player silencing
+		/// </summary>
+		private void EnablePlayerSilencing () {
+			if (!IsAirConsoleUnityPluginReady ()) {
+				throw new NotReadyException ();
```
Comment (daniel-n-dream) [inline Assets/AirConsole/scripts/AirConsole.cs:1949]: Is it normal that some functions are called with a space before the brackets and some don't? Is there no C# formatting guideline?

File: Assets/AirConsole/scripts/AirConsole.cs [hunk 4]
```diff
@@ -1930,6 +1939,26 @@ public class NotReadyException: SystemException {
...
+			}
+			JObject msg = new JObject {
+				{ "action", "enablePlayerSilencing" }
+			};
+
+			AllocateDeviceSlots (0);
```
Comment (daniel-n-dream) [inline Assets/AirConsole/scripts/AirConsole.cs:1955]: Where does that function live, I can't find it?

File: Assets/AirConsole/scripts/AirConsole.cs (hunk 3 above)
Comment (marc-n-dream) [reply Assets/AirConsole/scripts/AirConsole.cs:1949]: There are but not for spaces before squirmy brackets as there are different schools of thoughts on this in C languages. Same for opening squirmy on same line or not and else if 

from what I’ve seen there are certain official guidelines that were not followed (JavaScript casing and some other things, non C# file layout) while others were not consistently applied.

I will, if I have some headspace, see to configure and include Rider formatting that applies those things that don’t break API consistently on save.
Just takes time as C# has some 2xx formatting rules + Unity specificas (which only Riders engine even understands) and about 40 - 50 need to be enable/disabled, applied to the project and get verified.

File: Assets/AirConsole/scripts/AirConsole.cs (hunk 4 above)
Comment (marc-n-dream) [reply Assets/AirConsole/scripts/AirConsole.cs:1955]: What function are you referring to?
enablePlayerSilencing would be in api 1.9.0-alpha (branch of the same name in api) and as message in the js files you commented on before

Comment (daniel-n-dream) [reply Assets/AirConsole/scripts/AirConsole.cs:1955]: Sorry I meant the "AllocateDeviceSlots"

Comment (marc-n-dream) [reply Assets/AirConsole/scripts/AirConsole.cs:1955]: That is a function in AirConsole.cs, AirConsole class

**Comment by marc-n-dream** [comment]
All comments addressed, any more inputs @andrin-n-dream @daniel-n-dream?

**Review by daniel-n-dream — COMMENTED** [review]
LGTM

**Review by daniel-n-dream — APPROVED** [review]

**Pull request merged**